submissions/chris-voitova/frogger-game/app.js

```diff
@@ -0,0 +1,165 @@
+const field = {
+  width: 505,
+  borderX: 400,
+  borderY: 450,
+  waterCoord: 0,
+};
+
+const enemyProps = {
+  width: 98,
+  height: 56,
+};
+
+const playerProps = {
+  width: 70,
+  height: 86,
+  step: 50,
+};
+
+class Character {
+  constructor(positionX, positionY, width, height) {
+    this.positionX = positionX;
+    this.positionY = positionY;
+    this.width = width;
+    this.height = height;
+  }
+}
+
+// Enemies our player must avoid
+class Enemy extends Character {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+  constructor(
+    positionX,
+    positionY,
+    speed,
+    width = enemyProps.width,
+    height = enemyProps.height
+  ) {
+    super(positionX, positionY, width, height);
+
+    this.speed = speed;
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = "images/enemy-bug.png";
+  }
+  // Update the enemy's position, required method for game
+  // Parameter: dt, a time delta between ticks
+  update(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    this.positionX += this.speed * dt;
+    if (this.positionX > field.width) {
+      this.positionX = 0;
+    }
+  }
+  // Draw the enemy on the screen, required method for game
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.positionX, this.positionY);
+  }
+}
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+class Player extends Character {
+  constructor(
+    positionX,
+    positionY,
+    width = playerProps.width,
+    height = playerProps.height,
+    stepOnField = playerProps.step
+  ) {
+    super(positionX, positionY, width, height);
+
+    this.sprite = "images/char-boy.png";
+    this.initPosition = {
+      positionX: positionX,
+      positionY: positionY,
+    };
+    this.stepOnField = stepOnField;
+  }
+  checkCollisions(enemy) {
+    if (
+      enemy.positionX < this.positionX + this.width &&
+      enemy.positionX + enemy.width > this.positionX &&
+      enemy.positionY < this.positionY + this.height &&
+      enemy.positionY + enemy.height > this.positionY
+    ) {
+      alert("you are lose!");
+      this.goToInitPosition();
+    }
+  }
+  goToInitPosition() {
+    this.positionX = this.initPosition.positionX;
+    this.positionY = this.initPosition.positionY;
+  }
+  update() {
+    if (this.positionY <= field.waterCoord) {
```

Comment:
Move this check into `checkCollisions`. `update` must do update of your player only.

Reply:
moved this to checkWin method, because checkCollision called "number of enemies" times